projects/gateway/pkg/services/k8sadmisssion/validating_admission_webhook.go

```diff
@@ -241,28 +247,21 @@ func (wh *gatewayValidationWebhook) makeAdmissionResponse(ctx context.Context, r
 
 	proxyReports, validationErr := wh.validate(ctx, gvk, ref, req.Object, isDelete)
 
-	success := &v1beta1.AdmissionResponse{
-		Allowed: true,
-	}
-
-	if validationErr == nil {
-		logger.Debug("Succeeded")
+	isUnmarshalErr := validationErr != nil && errors.Is(validationErr, UnmarshalErr)
 
+	// even if validation is set to always accept, we want to fail on unmarshal errors
+	if !isUnmarshalErr && (validationErr == nil || wh.alwaysAccept) {
```

Comment:
nit:

```suggestion
	if validationErr == nil || (wh.alwaysAccept && !isUnmarshalErr)
```

a little easier to read imo